### ISSUE TITLE
Set HTTPUserAgent option

### ISSUE THREAD

**Issue by krlmlr** [rstudio/renv#520]
when installing from RSPM.

Otherwise we need to adapt `.Rprofile` -- easy to miss.

**Comment by kevinushey** [comment]
Does RSPM document what is expected for the user agent string?

**Comment by krlmlr** [comment]
In https://docs.rstudio.com/rspm/1.0.12/admin/binaries.html#binaries-r-configuration, section 17.3.3.

On my system and in my projects I'm using:

```r
options(HTTPUserAgent = sprintf("R/%s R (%s)", getRversion(), paste(getRversion(), R.version$platform, R.version$arch, R.version$os)))
```

**Comment by krlmlr** [comment]
Thanks for implementing. The user agent string in that commit looks slightly different from the recommended one, in particular the architecture is missing. Do you have a reference?

**Comment by kevinushey** [comment]
I don't; in that case I was just trying to mimic the default user agent provided by R. I've made a change to update the user agent string, although note that we only set it if `HTTPUserAgent` is already unset.

Do you think `renv` should more aggressively try to take over the user agent string here?

**Comment by krlmlr** [comment]
I have no idea, maybe @slopp can shed some light?

**Comment by slopp** [comment]
Yes, you should aggressively take over the user agent if you are hoping to get binaries from RSPM. The reason for this is a "feature" implemented in R > 3.6 where if the user agent is the default, than it gets removed entirely and substituted with the curl default when using libcurl. That is why RSPM's admin guide is adamant that you over-ride the default user agent with a slightly different user agent - that way the R information (like R version) is preserved regardless of the install method. We tried to convince R core that the "feature" was actually a bad move.

**Comment by krlmlr** [comment]
This seems to work reliably now, closing.
